### PR TITLE
coverage: Add UI tests for values accepted by `-Cinstrument-coverage`

### PR DESCRIPTION
**Pull request by Zalathar**
I wanted to clean up the code in `parse_instrument_coverage`, but it occurred to me that we currently don't have any UI tests for the various stable and unstable values supported by this flag.

---

Normally it might be overkill to individually test all the different variants of `on`/`off`, but in this case the parsing of those values is mixed in with some other custom code, so I think it's worthwhile being thorough.

### CODE REVIEW

**Comment by Zalathar** [comment]
@rustbot label +A-code-coverage

cc @Swatinem as this is relevant to #116094

Also indirectly relevant to the possible removal of `-Zinstrument-coverage` (#116980).

**Comment by Zalathar** [comment]
I would also like to change the documented default value from `all` to `on` (to permit `all` to possibly do something different in the future), but I'll propose and implement that separately.

File: src/tools/tidy/src/ui_tests.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ use std::path::{Path, PathBuf};
 const ENTRY_LIMIT: usize = 900;
 // FIXME: The following limits should be reduced eventually.
 const ISSUES_ENTRY_LIMIT: usize = 1854;
-const ROOT_ENTRY_LIMIT: usize = 865;
+const ROOT_ENTRY_LIMIT: usize = 866;
```
Comment (Zalathar) [inline src/tools/tidy/src/ui_tests.rs:14]: Bumping this by 1 because I'm adding a new directory for a feature (`-Cinstrument-coverage`) that I believe currently has no standalone UI tests.

Comment (Zalathar) [inline src/tools/tidy/src/ui_tests.rs:14]: Bumping this by 1 because I'm creating a new directory, and I'm not aware of any existing tests directly in `ui/` that could be moved into it.

**Comment by Zalathar** [comment]
Ah, looks like I need to add `// needs-profiler-support`.

(Or I could avoid flags that avoid needing the profiler runtime in the first place, but that doesn't seem necessary here.)

File: src/tools/compiletest/src/runtest.rs [hunk 2]
```diff
@@ -2470,7 +2470,7 @@ impl<'test> TestCx<'test> {
             }
             CoverageMap => {
                 rustc.arg("-Cinstrument-coverage");
-                // These tests only compile to MIR, so they don't need the
+                // These tests only compile to LLVM IR, so they don't need the
```
Comment (Zalathar) [inline src/tools/compiletest/src/runtest.rs:2473]: This is a mostly-unrelated comment fix that I happened to notice at around the same time.

**Review by Swatinem — APPROVED** [review]

**Comment by Zalathar** [comment]
Added an explicit error-case test for `-Cinstrument-coverage=` with no value [(diff)](https://github.com/rust-lang/rust/compare/6a3393ce76374953c8b3a4d2ebb2605d2e2fd987..71b7322440aaba17a163bcb962e5c8cb07239725).
